# Week 1 673380287-8 นายพีรพล แก้วเจริญสันติสุข

Lab 1.1: Environment Setup & First Script

In [50]:
print("Hello, Python Course!")

Hello, Python Course!


Lab 1.2: Working with Variables and Basic Data Types
- Objective: Students will define variables of different data types, perform simple operations, and display results.


2. Declare and print variables:


In [51]:
Age = 19
GPA = 3.51
Favorite_Color = "Blue"
student = True

print("Age = ",Age ,type(Age))
print("GPA = ",GPA,type(GPA))
print("Favorite_Color = ",Favorite_Color,type(Favorite_Color))
print("Student = ",student,type(student))

Age =  19 <class 'int'>
GPA =  3.51 <class 'float'>
Favorite_Color =  Blue <class 'str'>
Student =  True <class 'bool'>


3. Perform simple arithmetic:


In [52]:
num1 = 15
num2 = 3
sum = num1 + num2


different = num1 - num2
product = num1 * num2
quotient = num1 / num2

print("num1 = ",num1)
print("num2 = ",num2,)

print("Sum = ",sum)
print("Different = ",different)
print("Product = ",product)
print("Quotient = ",quotient)

num1 =  15
num2 =  3
Sum =  18
Different =  12
Product =  45
Quotient =  5.0


4. String concatenation and f-strings:


In [53]:
First_name = "Peerapol"
Last_name = "Kaewcharoensontisuk"
Full_name = First_name + " " + Last_name
print(f"Hello,{Full_name}")

Hello,Peerapol Kaewcharoensontisuk


5. User input and type conversion:


In [54]:
num = input("What your favorite number : ")
print(f"Your favorite number is {num} (type: {type(num).__name__})")
try:
    num_int = int(num)
    print(num_int," + 5 = ", num_int+5)
except ValueError:
    print("Invalid input. Please enter an Number.")




What your favorite number : 10
Your favorite number is 10 (type: str)
10  + 5 =  15


Activity 1.1: Class Discussion - Uses of Scripting
Format: Open discussion, possibly in small groups first, then full class.
Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."
Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.


**คำตอบ:** สามารถนำไปเขียนประยุกตร์ใช้กับการอ่านข้อมูลจากไฟล์ต่างๆที่มีจำนวนมากได้ เช่น อ่านข้อมูลจาก excel หรือ ข้อมูลที่มีความยาวมากๆ เราก็สามารถเขียน python ในการวนอ่านซ้ำๆเก็บข้อมูลที่เราต้องการได้ นอกจากนี้ยังสามารถนำไปประมวลผลรูปภาพ สร้างกราฟ ข้อมูลเชิงสถิติ แก้ไขปัญหาคณิตศาสตร์ต่างๆได้อีกด้วย

Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [55]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))


13.5
<class 'float'>


ได้ผลลัพธ์: 13.5

 ประเภทdatatype เป็น: float

ทำไมถึงเป็นtype float : เพราะเมื่อนำข้อมูลที่มี datatype เป็น int มาบวกกับ float ภาษา Python จะเปลียนเป็น float โดยอัตโนมัติ

In [56]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


ได้ผลลัพธ์: Alice is 30 years old.

ประเภทdatatype เป็น: String

ทำไมถึงเป็นtype String : เมื่อเราสร้างตัวแปรอะไรก็ตามแล้วใส่เครื่อหมาย "" คลุมค่าตัวแปรนั้นไว้ ภาษา Python จะกำหนดให้ค่านั้นเป็น String เสมอ

In [57]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


ได้ผลลัพธ์: False

ประเภทdatatype เป็น: Boolean

ทำไมถึงเป็นtype boolaean : เมื่อเราประกาศชื่อตัวแปรแล้ว หลัง = เป็น True หรือ False ภาษา Python จะมองว่าตัวแปรพวกนี้มี datatype เป็น Boolean และเมื่อนำตัวแปรทั้งสองมากระทำกันด้วย and ก็จะได้ค่า boolean

In [58]:
# Puzzler 4
# What happens here?
# num_str = input("Enter a number: ")
# result = num_str + 5
# print(result)

ได้ผลลัพธ์: ไม่มีผลลัพธ์ออกมา

ประเภทdatatype เป็น: ไม่มี

ทำไมถึงเป็นtype ไม่มี : เพราะเมื่อใส่ # หน้าโค๊ดบรรทัดที่เขียนไป Python จะกำหนดให้บรรทัดนั้นเป็นคอมเม้นทั้งหมด